lib/ansible/plugins/inventory/tower.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: tower
+    plugin_type: inventory
+    authors:
+      - Matthew Jones (@matburt)
+      - Yunfan Zhang <yunfzhan@redhat.com>
+    short_description: Ansible dynamic inventory plugin for Ansible Tower.
+    description:
+        - Imports inventories from Ansible Tower.
+        - Reads configurations from environment variables.
+        - You need to specify TOWER_HOST (the network address of your Ansible Tower host),
+          TOWER_USERNAME (the user you plan to use to access inventories on tower),
+          TOWER_PASSWORD (password for TOWER_USERNAME),
+          and TOWER_INVENTORY (the ID of the tower inventory that you wish to import).
+        - The path for configuration file should always be tower_inventory.
+        - The plugin will read inventory specified by TOWER_INVENTORY from Ansible tower.
+'''
+
+EXAMPLES = '''
+# Before you execute the following commands, you should make sure this file is in your plugin path,
+# and you enabled this plugin.
+# Set environment variables:
+# export TOWER_HOST=YOUR_TOWER_HOST_ADDRESS
```

Comment:
Rather than limiting this to environment variables you could utilize config to allow a more flexible use case. That would let you document each option individually and mark them as required so you don't have to manually check in the plugin that they were provided.
e.g. 
* document options with their environment components:
```
options:
    tower_host:
        description: ...
        env: [TOWER_HOST]
        required: true
    ...
```
* allow parsing of a file called tower_inventory by calling self._read_config_data(path)
* access those options with self.get_option('tower_host') after either calling self._read_config_data(path) (which will merge options provided in the tower_inventory file with the environment vars in a self._options dict)

Reply:
you could keep the use case of no inventory config file if the path doesn't exist by calling self.set_options yourself (rather than happening in _read_config_data()) to have access to the environment variables with get_option()

Reply:
Thanks @s-hertel ! I will make these modification.